### DISCUSSION TITLE
Search for unstructured phone numbers inside of a supabase table

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#2906] [Questions] [answered]
Hello team,

I have an array of phone numbers which is pulled in from the phone contacts. Some of them have country codes on them and some don't. I am trying to search for phone contacts that is available in the phone number column of the supabase table which is in a structured format, which means there are country codes on all of them and there are no spaces or special characters on them. 

I am trying to make this possible in one single query, trying to minimize on the number of API calls.

Couple ways I thought about this was -

1. Use the **in** operator. With this I wont be able to find any phone numbers that don't have country codes in them. 
   As of now it gives me this error message -   **'in' can't be used as an identifier because it's a keyword.
Try renaming this to be an identifier that isn't a keyword.**  
    final response = await supabase
        .from('users')
        .select('*')
        .in('phone_number', ['phone_array'])
        .execute();

2. Use the **like** operator inside of a for loop and do a manual search for every single phone number in the database. This could result in a massive amount of API calls, which I don't want.
    final response = await supabase
        .from('users')
        .select('*')
        .like('phone_number', '%6720000000')
        .execute();

Best advise on the most efficient way I can accomplish this.

**Comment by steve-chavez** [comment]
I think a [computed column](https://postgrest.org/en/stable/api.html#computed-columns) might help here. Check this previous discussion: https://github.com/supabase/supabase/discussions/2415#discussioncomment-1023363

**Reply by sumchans** [reply]
@steve-chavez not sure how a computed column would work in this case.

**Comment by steve-chavez** [comment]
>'in' can't be used as an identifier because it's a keyword. Try renaming this to be an identifier that isn't a keyword.

Strange, do you have an `in` value passed to the filter somewhere in the array?

**Reply by sumchans** [reply]
@steve-chavez just curious to know, how would you find a solution for this. Every way I think about it is not going to be 100% effective. I can do a search on the phone number column using the like operator, but I read that there could be duplicate phone numbers if no country code is added to them. Also I thought of extracting the last 10 digits of the phone number and then do a search on it, but I read that depending on the country, phone numbers could be 0-15 digits long.

**Comment by steve-chavez** [comment, marked as answer]
Seems this one was already answered on: https://github.com/supabase/supabase/discussions/2938
